### ISSUE TITLE
ginkgo: known test flakes 

### ISSUE THREAD

**Issue by aalemayhu** [cilium/cilium#2291]
Adding this issue here to keep track of some of the current flakes seen. I think the ones I have seen are just in the multiple monitor output and the policy enforcement. Will walk through the last builds and add comments to this.

**Comment by aalemayhu** [comment]
Build 437, https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/437/

The panic in the monitor test is probably that we are checking a string in the test and get an out of bounds access.
```
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/monitor.go:202
Test Panicked
/usr/local/go/src/runtime/panic.go:491
```

For the rest I don't know enough about the policy test there but including all of them below.

```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:123
Expected
    <int>: 0
to equal
    <int>: 1
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:139
Standard Ausgabe (STDOUT)

STEP: Check no policy enforcement
STEP: Setting to Always
```

```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:153
Expected
    <int>: 0
to equal
    <int>: 1
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:170
```

```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:246
Expected
    <int>: 0
to equal
    <int>: 1
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:250
```
```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:262
Expected
    <int>: 0
to equal
    <int>: 1
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:265
```
```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:297
Expected
    <int>: 0
to equal
    <int>: 1
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/Policies.go:300
```

**Comment by aalemayhu** [comment]
Build 436, https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/436/#showFailuresLink

Monitor line issues

```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/monitor.go:128
Expected
    <int>: 2
to be >
    <int>: 3
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/monitor.go:143
```
Skipping some of the already mentioned policy ones above.

**Comment by aalemayhu** [comment]
Build 435, https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/435/#showFailuresLink

The newest one on this list is the Kafka one. No monitor test failure in this one but the policy ones from above are there.

```
Stacktrace

/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/kafka.go:112
Expected
    <string>: Message 1
    Message 2
    Message 3
    Message 4
    
to contain substring
    <string>: Processed a total of 5 messages
/root/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/src/github.com/cilium/cilium/test/runtime/kafka.go:139
Standard Ausgabe (STDOUT)

STEP: Allowed topic
```

**Comment by eloycoto** [comment]
Hey @scanf the policies ones are fixed on PR https://github.com/cilium/cilium/pull/2284

Regards

**Comment by aalemayhu** [comment]
@eloycoto Thanks for the heads up.

**Comment by manalibhutiyani** [comment]
Assigning to Ian, Please reassign if needed.

**Comment by eloycoto** [comment]
Hi @scanf, 

I think that this issue can be closed: 

- The Kafka issue was with the kernel 4.10. I downgrade the image to 4.9 Kernel and correct LLVM, and it didn't fail in the last ten builds. 

- The issue with Multiple monitors was fixed on PR #2381 

- The issue around policies was fixed on PR #2284 

So, I think that this issue can be closed. At the moment the master is failing, but it's fixed on PR #2388

**Comment by aalemayhu** [comment]
Sounds good. Thanks for following up!
